workers/unity/Packages/com.improbable.gdk.core/Systems/SpatialOSReceiveSystem.cs

```diff
@@ -81,24 +77,62 @@ protected override void OnUpdate()
 
         private void OnAddEntity(AddEntityOp op)
         {
-            view.CreateEntity(op.EntityId);
+            var entityId = op.EntityId;
+            if (worker.EntityMapping.ContainsKey(entityId))
+            {
+                worker.LogDispatcher.HandleLog(LogType.Error, new LogEvent(Errors.DuplicateAdditionOfEntity)
+                    .WithField(LoggingUtils.LoggerName, LoggerName)
+                    .WithField(LoggingUtils.EntityId, entityId));
+                return;
+            }
+
+            var entity = EntityManager.CreateEntity();
+            EntityManager.AddComponentData(entity, new SpatialEntityId
+            {
+                EntityId = entityId
+            });
+            EntityManager.AddComponentData(entity, new NewlyAddedSpatialOSEntity());
+
+            AddAllCommandComponents.ForEach(action => action(entity));
```

Comment:
the name seems fairly self-explanatory. I can add a comment, if you want, but it would be something like

`Adding all command senders and responders to the entity`

Reply:
In this case, avoiding LINQ and using a simple `foreach` may be worthwhile, for clarity and avoiding unneeded LINQ-y allocations.
(I know I recommend moving from `foreach` to LINQ most times, but moving from `foreach` -> `ForEach` is a bit of a smell, IMO.)

Reply:
done